app/src/main/java/org/schabi/newpipe/info_list/InfoItemDialog.java

```diff
@@ -1,54 +1,296 @@
+ * Dialog for a {@link StreamInfoItem}.
+ * The dialog'S content are actions that can be performed on the {@link StreamInfoItem}.
+ * This dialog is mostly used for longpress context menus.
+ */
+public final class InfoItemDialog {
+    /**
+     * Ideally, {@link InfoItemDialog} would extend {@link AlertDialog}.
+     * However, extending {@link AlertDialog} requires many additional lines
+     * and brings more complexity to this class, especially the constructor.
+     * To circumvent this, an {@link AlertDialog.Builder} is used in the constructor.
+     * Its result is stored in this class variable to allow access via the {@link #show()} method.
+     */
+    private final AlertDialog dialog;
 
-    public InfoItemDialog(@NonNull final Activity activity,
-                          @NonNull final String[] commands,
-                          @NonNull final DialogInterface.OnClickListener actions,
-                          @NonNull final String title,
-                          @Nullable final String additionalDetail) {
+    private InfoItemDialog(@NonNull final Activity activity,
+                           @NonNull final Fragment fragment,
+                           @NonNull final StreamInfoItem info,
+                           @NonNull final List<StreamDialogEntry> entries) {
 
         final View bannerView = View.inflate(activity, R.layout.dialog_title, null);
         bannerView.setSelected(true);
 
         final TextView titleView = bannerView.findViewById(R.id.itemTitleView);
-        titleView.setText(title);
+        titleView.setText(info.getName());
 
         final TextView detailsView = bannerView.findViewById(R.id.itemAdditionalDetails);
-        if (additionalDetail != null) {
-            detailsView.setText(additionalDetail);
+        if (info.getUploaderName() != null) {
+            detailsView.setText(info.getUploaderName());
             detailsView.setVisibility(View.VISIBLE);
         } else {
             detailsView.setVisibility(View.GONE);
         }
 
+        final String[] items = entries.stream()
+                .map(entry -> entry.getString(activity)).toArray(String[]::new);
+
+        final DialogInterface.OnClickListener action = (d, index) ->
+            entries.get(index).action.onClick(fragment, info);
+
         dialog = new AlertDialog.Builder(activity)
                 .setCustomTitle(bannerView)
-                .setItems(commands, actions)
+                .setItems(items, action)
                 .create();
+
     }
 
     public void show() {
         dialog.show();
     }
+
+    /**
+     * <p>Builder to generate a {@link InfoItemDialog}.</p>
+     * Use {@link #addEntry(StreamDialogDefaultEntry)}
+     * and {@link #addAllEntries(StreamDialogDefaultEntry...)} to add options to the dialog.
+     * <br>
+     * Custom actions for entries can be set using
+     * {@link #setAction(StreamDialogDefaultEntry, StreamDialogEntry.StreamDialogEntryAction)}.
+     */
+    public static class Builder {
```

Comment:
In the Javadoc I'd explain that it can only be used to show a dialog for `StreamInfoItem`s, not just any `InfoItem`

Reply:
I thought about renaming the whole class to `StreamInfoItemDialog` to  make it clear that the dialog only works with `StreamInfoItems` currently.
 On the other hand, we could create more builders to match the missing InfoItems: channels and playlists. Their `held()` methods are not overridden in many cases.